lib/shallow.ts

```diff
@@ -5,7 +5,10 @@ import { RecursivePartial } from './models/recursive-partial';
 import { InvalidStaticPropertyMockError, Renderer } from './models/renderer';
 import { Rendering, RenderOptions } from './models/rendering';
 import { TestSetup } from './models/test-setup';
+
+// Side effect modules
 import './test-framework';
+import './tools/intercept-injectable';
```

Comment:
I sort-of feel like intercepting providedIn modules should be explicit. The issue is that we MUST import `Shallow` before any other JS modules are interpreted or we won't properly intercept the `@Injectable` decorator.  This could be very tricky to figure out because if I'm writing a test and I import my component or service BEFORE I import Shallow, my root provider will be captured.

I'm thinking we should force this feature to be "opt-in" so they will have to set it up in their karma shim (test-setup) file:

Maybe:
```typescript
// In the global test-setup
import {Shallow} from 'shallow-render';
Shallow.interceptProviders();
// ...
```